src/HOC/withAuth.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import * as React from 'react';
+import { connect } from 'react-redux';
+import type { StoreType } from '../store/reducers';
+import type { State as SessionState } from '../store/reducers/SessionReducer';
+import { saveToken, signIn } from '../store/actions/SessionActions';
+import { getCookieToken } from '../helpers/Auth';
+
+type Props = {
+  doSignIn: Function,
```

Comment:
I would be more specific about the function types => e.g. `doSaveToken: (token: string) => void`